### PR TITLE
multi: fix lint issues.

### PR DESCRIPTION
**Pull request by dnldd**
This fixes lint outstanding lint issues and wraps errors created by the postgres db driver in order to leverage errors.Is/As functionality.

### CODE REVIEW

**Review by dajohi — APPROVED** [review]

File: errors/error.go [hunk 1]
```diff
@@ -27,12 +27,27 @@ const (
...
+	DBClose = ErrorKind("ErrDBClose")
+
 	// DBUpgrade indicates a database upgrade error.
 	DBUpgrade = ErrorKind("ErrDBUpgrade")
 
 	// PersistEntry indicates a database persistence error.
 	PersistEntry = ErrorKind("ErrPersistEntry")
 
+	// CreateTable indicates a database table creation error.
+	CreateTable = ErrorKind("CreateTable")
```
Comment (jholdstock) [inline errors/error.go:40]: CreateTable is specific to SQL and does not apply to Bolt. As we are aiming to make our DB interface agnostic of the implementation, the interface should not include implementation specific errors. 

This also applies to some existing errors - eg. `ErrBucketCreate`  does not apply to SQL, only Bolt.

We should combine these into something more generic, perhaps something like `ErrCreateStorage`

File: errors/error.go [hunk 2]
```diff
@@ -27,12 +27,27 @@ const (
...
+
+	// BeginTx indicates a database transaction begin error.
+	BeginTx = ErrorKind("BeginTx")
```
Comment (jholdstock) [inline errors/error.go:43]: All of these tx errors can probably just be `ErrPersistEntry`. Right now, the calling doesn't need to know the precise nature of the failure, its enough just to know that something went wrong. The nested error generated by the SQL library will contain more details for the log file.

**Review by jholdstock — CHANGES_REQUESTED** [review]

File: errors/error.go [hunk 3]
```diff
@@ -16,119 +16,122 @@ const (
 	// ------------------------------------------
 
 	// ValueNotFound indicates no value found.
-	ValueNotFound = ErrorKind("ErrValueNotFound")
+	ValueNotFound = ErrorKind("ValueNotFound")
 
 	// BucketNotFound indicates no bucket found.
-	BucketNotFound = ErrorKind("ErrBucketNotFound")
+	BucketNotFound = ErrorKind("BucketNotFound")
```
Comment (jholdstock) [inline errors/error.go:22]: Maybe rename to `StorageNotFound`?

File: pool/postgres.go [hunk 4]
```diff
@@ -44,53 +44,66 @@ func InitPostgresDB(host string, port uint32, user, pass, dbName string) (*Postg
 		return nil, errs.DBError(errs.DBOpen, desc)
 	}
 
+	makeErr := func(funcName string, table string, err error) error {
```
Comment (jholdstock) [inline pool/postgres.go:47]: passing `funcName` as a param is unnecessary - it is already in scope for the body of this func.

File: pool/postgres.go [hunk 5]
```diff
@@ -219,7 +253,9 @@ func (db *PostgresDB) fetchCSRFSecret() ([]byte, error) {
 			return nil, errs.DBError(errs.ValueNotFound, desc)
 		}
 
-		return nil, err
+		desc := fmt.Sprintf("%s: unable to fetch CSRF secret: %v",
+			funcName, err)
+		return nil, errs.DBError(errs.PersistEntry, desc)
```
Comment (jholdstock) [inline pool/postgres.go:258]: We're fetching here, not persisting

File: pool/postgres.go [hunk 6]
```diff
@@ -234,28 +270,51 @@ func (db *PostgresDB) fetchCSRFSecret() ([]byte, error) {
...
 	tx, err := db.DB.Begin()
 	if err != nil {
 		return err
 	}
 
 	_, err = tx.Exec(insertLastPaymentHeight, height)
 	if err != nil {
-		tx.Rollback()
-		return err
+		rErr := tx.Rollback()
```
Comment (jholdstock) [inline pool/postgres.go:294]: Nice catch finding this, I didnt notice that `Rollback` could return an error.

I'm not sure about the way it is handled though. If we have a `tx.Exec` error and a `tx.Rollback` error, only the `Rollback` error is being returned and logged. The `Exec` error, which could be the root cause of the issue, will be lost. I think we need to log both.

File: pool/postgres.go [hunk 7]
```diff
@@ -234,28 +270,51 @@ func (db *PostgresDB) fetchCSRFSecret() ([]byte, error) {
...
+		if rErr != nil {
+			desc := fmt.Sprintf("%s: unable to rollback persisting last "+
+				"payment height tx: %v", funcName, rErr)
+			return errs.DBError(errs.PersistEntry, desc)
+		}
+
+		desc := fmt.Sprintf("%s: unable to persist last payment height: %s",
+			funcName, err.Error())
```
Comment (jholdstock) [inline pool/postgres.go:302]: Why use `"%s", err.Error()` here and in the following instance, when `"%v", err` is used everywhere else?

File: pool/postgres.go [hunk 8]
```diff
@@ -353,15 +424,22 @@ func (db *PostgresDB) fetchAccount(id string) (*Account, error) {
...
 	return &Account{uuid, address, createdOn}, nil
 }
 
 // deleteAccount purges the referenced account from the database.
 func (db *PostgresDB) deleteAccount(id string) error {
+	const funcName = "deleteAccount"
 	_, err := db.DB.Exec(deleteAccount, id)
-	return err
+	if err != nil {
+		desc := fmt.Sprintf("%s: unable to delete account: %v", funcName, err)
```
Comment (jholdstock) [inline pool/postgres.go:439]: include account id?

File: pool/postgres.go [hunk 9]
```diff
@@ -353,15 +424,22 @@ func (db *PostgresDB) fetchAccount(id string) (*Account, error) {
...
+		return errs.DBError(errs.FetchEntry, desc)
```
Comment (jholdstock) [inline pool/postgres.go:440]: Deleting here, not fetching

File: pool/postgres.go [hunk 10]
```diff
@@ -406,24 +484,35 @@ func (db *PostgresDB) PersistPayment(p *Payment) error {
...
 }
 
 // deletePayment purges the referenced payment from the database. Note that
 // archived payments cannot be deleted.
 func (db *PostgresDB) deletePayment(id string) error {
+	const funcName = "deletePayment"
 	_, err := db.DB.Exec(deletePayment, id)
-	return err
+	if err != nil {
+		desc := fmt.Sprintf("%s: unable to delete payment: %v", funcName, err)
```
Comment (jholdstock) [inline pool/postgres.go:512]: include id?

File: pool/postgres.go [hunk 11]
```diff
@@ -432,13 +521,22 @@ func (db *PostgresDB) ArchivePayment(p *Payment) error {
...
+		return errs.DBError(errs.PersistEntry, desc)
 	}
 
 	_, err = tx.Exec(deletePayment, p.UUID)
 	if err != nil {
-		tx.Rollback()
-		return err
+		rErr := tx.Rollback()
+		if rErr != nil {
+			desc := fmt.Sprintf("%s: unable to rollback delete payment tx: %v",
```
Comment (jholdstock) [inline pool/postgres.go:533]: As above, we may lose the root cause error here

File: pool/postgres.go [hunk 12]
```diff
@@ -618,7 +758,9 @@ func (db *PostgresDB) fetchAcceptedWork(id string) (*AcceptedWork, error) {
 			return nil, errs.DBError(errs.ValueNotFound, desc)
 		}
 
-		return nil, err
+		desc := fmt.Sprintf("%s: unable to fetch accepted work: %v",
```
Comment (jholdstock) [inline pool/postgres.go:761]: id?

File: pool/postgres.go [hunk 13]
```diff
@@ -674,18 +820,26 @@ func (db *PostgresDB) updateAcceptedWork(work *AcceptedWork) error {
 
 // deleteAcceptedWork removes the associated accepted work from the database.
 func (db *PostgresDB) deleteAcceptedWork(id string) error {
+	const funcName = "deleteAcceptedWork"
 	_, err := db.DB.Exec(deleteAcceptedWork, id)
-	return err
+	if err != nil {
+		desc := fmt.Sprintf("%s: unable to delete account work: %v",
```
Comment (jholdstock) [inline pool/postgres.go:826]: id?

File: pool/postgres.go [hunk 14]
```diff
@@ -737,20 +895,33 @@ func (db *PostgresDB) persistJob(job *Job) error {
...
 	return nil
 }
 
 // deleteJob removes the associated job from the database.
 func (db *PostgresDB) deleteJob(id string) error {
+	const funcName = "deleteJob"
 	_, err := db.DB.Exec(deleteJob, id)
-	return err
+	if err != nil {
+		desc := fmt.Sprintf("%s: unable to delete job entry: %v",
```
Comment (jholdstock) [inline pool/postgres.go:909]: id?

**Review by jholdstock — CHANGES_REQUESTED** [review]

File: pool/postgres.go [hunk 15]
```diff
@@ -234,28 +270,53 @@ func (db *PostgresDB) fetchCSRFSecret() ([]byte, error) {
...
+		rErr := tx.Rollback()
+		if rErr != nil {
+			desc := fmt.Sprintf("%s: unable to rollback persist last payment "+
+				"paid on time tx: %v, initial error: %v", funcName,
+				rErr, err)
+			return errs.DBError(errs.PersistEntry, desc)
+		}
+
+		desc := fmt.Sprintf("%s: unable to persist last payment paid on "+
+			"time: %s", funcName, err.Error())
```
Comment (jholdstock) [inline pool/postgres.go:318]: `"%v", err`

File: pool/postgres.go [hunk 16]
```diff
@@ -449,29 +552,48 @@ func (db *PostgresDB) ArchivePayment(p *Payment) error {
 		aPmt.CreatedOn, aPmt.PaidOnHeight, aPmt.TransactionID, aPmt.Source.BlockHash,
 		p.Source.Coinbase)
 	if err != nil {
-		tx.Rollback()
-		return err
+		rErr := tx.Rollback()
+		if rErr != nil {
+			desc := fmt.Sprintf("%s: unable to rollback archived payment "+
```
Comment (jholdstock) [inline pool/postgres.go:557]: Missing initial error

**Review by jholdstock — CHANGES_REQUESTED** [review]
Two more things and this is good to go.

**Review by jholdstock — APPROVED** [review]

**Pull request merged**